Bindings/Java/Matlab/Utilities/osimList.m

```diff
@@ -69,25 +89,36 @@
             end
             outputnames = outputnames';
         end
-        function reference = get(obj,name)
+        function reference = getByName(obj,name)
             % return a reference for a object in the list
             import org.opensim.modeling.*
-            model = obj.model;
             list = obj.list;
             li = list.begin();
             while ~li.equals(list.end())
                 if strcmp(char(li.getName()),name)
-                    pathname = li.getAbsolutePathName();
-                    comp = model.getComponent(pathname);
-                    class = comp.getConcreteClassName();
-                    eval(['reference = ' char(class) '.safeDownCast(comp);'])
+                    reference = li.deref;
                     break
                 end
                 li.next();
             end 
             if ~exist('reference','var')
-                error(['Component name: ' name ' is not found'])
+                error(['Component name not found: ' name])
+            end
+        end
+        function reference = getByIndex(obj,index)
```

Comment:
FWIW, I would expect most MATLABers to use a `for` loop by default. To avoid inefficiency, might it be possible to keep track of the last item that was retrieved and start searching from there (essentially doing `li.next()` under the covers until `list.end()`, then starting from the beginning)? Alternatively, perhaps just documentation (with the above warning and code snippets) would be helpful under the `getByIndex()` method.

Reply:
Thanks @tkuchida, @aseth1, and @chrisdembia for the awesome comments! This design feedback is great— I have been spending some time thinking about the philosophy of the list design and how that conflicts with the typical Matlab user for-loop strategy. The original reason I wanted to make this was to make it easier for me to get a single component from a list without having to write a while loop everytime. Perhaps it would be better to simplify this class to a set of static methods that are for single use only;

```MATLAB
% Get the number of components in the model
nComps = osimList.getNumberOfComponents(model, 'Body')
% Get all the body names (as strings)
bNames = osimList.getNames(model, 'Body') 
% I only want to work on the Pelvis, so I get it directly. 
pelvis = osimList.getComponent(model, 'Body', names{i} ) 
```
This wouldn't stop anyone from using it in a for-loop, but we could write in documentation when to use the above method and when to use the list directly (when you want to deal with most components in the list).

I am happy to jump on a call to discuss since I feel this may need discussion.